client/state/purchases/selectors.js

```diff
@@ -21,7 +24,7 @@ export const getPurchasesError = state => state.purchases.error;
  * @return {Object} the matching purchase if there is one
  */
 export const getByPurchaseId = ( state, purchaseId ) => (
-	getPurchases( state ).filter( purchase => purchase.id === purchaseId ).shift()
+	purchasesAssembler.createPurchasesArray( getPurchases( state ).filter( purchase => purchase.ID === purchaseId ) ).shift()
```

Comment:
Something I noticed when working on #6647 is that I think we should be assembling the result of `getSitePurchases` and `getUserPurchases`. We can do this by either:

- Moving this `createPurchasesArray` to `getPurchases`. This will slow down `getByPurchaseId`, but it might not matter if we start caching the result of these selectors with `createSelector`.
- Adding it to both `getUserPurchases` and `getSitePurchases`.